superset/assets/src/components/AnchorLink.jsx

```diff
@@ -44,20 +47,32 @@ class AnchorLink extends React.PureComponent {
     const { anchorLinkId } = this.props;
 
     if (hash && anchorLinkId === hash) {
-      const directLinkComponent = document.getElementById(anchorLinkId);
-      if (directLinkComponent) {
-        setTimeout(() => {
-          directLinkComponent.scrollIntoView({
-            block: 'center',
-            behavior: 'smooth',
-          });
-        }, 1000);
-      }
+      this.scrollToView();
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { inFocus = false } = nextProps;
+    if (inFocus) {
+      this.scrollToView(0);
     }
   }
 
   getLocationHash() {
```

Comment:
thanks. i added this instance method before, because I had difficulty to add unit test, which needs to mock window.location.hash. now i found a better solution here.